### ISSUE TITLE
Ensure proper branch protection for `main`.

### ISSUE THREAD

**Issue by Leon-JS02** [mehrdadhalali/Games-Release-Tracker#7]
As a data engineer and contributor to this project, I would like the `main` branch of the repo to be protected.
This should involve
- Ensuring that merging directly into main is blocked
- Ensure that pull requests into the main branch requires at least 2 approvals from collaborators before merging

This will ensure that contributions need to be approved before being merged into main, preventing potential issues and merge conflicts further down the line for other developers.
